pipelines/src/main/scala/dagr/pipelines/ParallelHaplotypeCaller.scala

```diff
@@ -101,14 +60,29 @@ class ParallelHaplotypeCaller
  @arg(flag="t", doc="Temporary directory in which to store intermediate results.") val tmpDirectory: Option[Path],
  @arg(flag="H", doc="Use the native code for GATK (Unix only)")                    val useNativePairHmm: Boolean = false
 )
-extends PathBasedScatterGatherPipeline(
-  domain                      = ref,
-  output                      = output,
-  splitInputPathTaskGenerator = (tmpDir: Option[Path]) => (ref: Path)             =>
-    new SplitIntervalsForCallingRegions(ref=ref, intervals=intervals, output=Some(ParallelHaplotypeCaller.getTempDirectory(tmpDir, "intervals_for_calling.")), maxBasesPerScatter=maxBasesPerScatter),
-  scatterTaskGenerator        = (tmpDir: Option[Path]) => (intv: PathToIntervals) =>
-    new ScatteredSingleSampleHaplotypeCaller(ref=ref, bam=input, filterVcf=filterVcf, dbsnp=dbsnp, tmpDirectory=tmpDir, intervals=intv, useNativePairHmm=useNativePairHmm),
-  gatherTaskGenerator         = (in: Seq[PathToVcf], out: PathToVcf)              =>
-    new GatherVcfs(in=in, out=out) with GatherTask[PathToVcf] { def gatheredOutput: PathToVcf = out },
-  tmpDirectory                = tmpDirectory
-)
+extends Pipeline(outputDirectory = Some(output.getParent)) {
+  private def replace(p: Path, a: String, b: String): Path = p.getParent.resolve(p.getFileName.toString.replace(a, b))
+
+  override def build(): Unit = {
+    val dir = tmpDirectory match {
+      case Some(p) => Files.createTempDirectory(p, "parallel_hap")
+      case None    => Files.createTempDirectory("parallel_hap")
+    }
+
+    val scatterer = new SplitIntervalsForCallingRegions(ref=ref, intervals=intervals, output=Some(dir), maxBasesPerScatter=maxBasesPerScatter)
+    val scatter = Scatter(scatterer)
+
+    val hc = scatter.map(il => new HaplotypeCaller(ref=ref, intervals=Some(il), bam=input, vcf=PathUtil.replaceExtension(il, ".g.vcf.gz"), useNativePairHmm=useNativePairHmm))
+    val gt = hc.map(h => GenotypeGvcfs(ref=ref, intervals=h.intervals, gvcf=h.vcf, vcf=replace(h.vcf, ".g.vcf.gz", ".vcf.gz"), dbSnpVcf = dbsnp ))
+    val gather = if (filterVcf) {
+      val filter = gt.map(g => new FilterVcf(in=g.vcf, out=replace(g.vcf, ".vcf.gz", ".filtered.vcf.gz")))
+      filter.gather(fs => new GatherVcfs(in=fs.map(_.out), out=output))
+    }
+    else {
+      gt.gather(gts => new GatherVcfs(in=gts.map(_.vcf), out=output))
+    }
+
+    root ==> scatter
+    gather ==> new DeleteFiles(dir)
```

Comment:
I think this would be fine - though actually it would be:
```scala
root ==> scatter ==> hc ==> gt ==> gather ==> new DeleteFiles(dir)
```
It's just that `scatter ==> hc ==> gt ==> gather` is auto-wired by the scatter/gather mechanics.